docs/3.x.x/guides/filters.md

```diff
@@ -84,54 +115,88 @@ Get the second page of results.
-
-::: note
-The returned objects are formatted according to the ORM used by the model.
-:::
-
-#### Example
-
-**Path —** `./api/user/controllers/User.js`.
+To transform the query params to Strapi's standard filters a request, you can use the `convertRestQueryParams` function from [strapi-utils](../api-reference/reference.md#strapiutils).
 
 ```js
-// Define a list of params.
-const params = {
-  '_limit': 20,
-  '_sort': 'email'
-};
+const { convertRestQueryParams } = require('strapi-utils');
 
-// Convert params.
-const formattedParams = strapi.utils.models.convertParams('user', params); // { limit: 20, sort: 'email' }
+module.exports = {
+  // Find products having a price equal or greater than `3`.
+  fetchExpensiveProducts: (params, populate) => {
+    const filters = convertRestQueryParams(params);
+
+    /**
+     *  filters = {
+     *    start: 0,
+     *    limit: 10,
+     *    sort: [{ field: 'id', order: 'desc' }],
+     *    where: [
+     *      { field: 'id', operator: 'eq', value: 1 },
+     *    ]
+     * }
+     */
+
+    // do sth with them
+  },
+};
 ```
 
-### Query usage
-
-#### Example
+### Querying data
 
-**Path —** `./api/user/controllers/User.js`.
+We added a new API to query data base on the new filters API.
 
 ```js
-module.exports = {
+const { convertRestQueryParams, buildQuery } = require('strapi-utils');
 
+module.exports = {
   find: async (ctx) => {
     // Convert params.
-    const formattedParams = strapi.utils.models.convertParams('user', ctx.request.query);
-
-    // Get the list of users according to the request query.
-    const filteredUsers = await User
-      .find()
-      .where(formattedParams.where)
-      .sort(formattedParams.sort)
-      .skip(formattedParams.start)
-      .limit(formattedParams.limit);
-
-    // Finally, send the results to the client.
-    ctx.body = filteredUsers;
+    const filter = convertRestQueryParams(ctx.request.query);
+
+    return buildQuery({
+      model: Article
+      filters,
+      populate: []
+    });
   };
 };
 ```
+
+
+#### SQL databases (strapi-hook-bookshelf)
+
+If you are using a SQL database, calling `buildQuery` will return a [`Bookshelf Query`](https://bookshelfjs.org/api.html) on which you can call other functions (e.g `count`)
+
+#### Mongo database
+
+If you are using a mongo database calling `buildQuery` returns either a [`Mongoose Query`](https://mongoosejs.com/docs/api.html#Query) or a custom query when used with deep filtering.
+
+
+
+##### Custom Query
+
+When using the deep filtering feature with mongo, we build an aggregation query to avoid too many round-trips with the mongo DB.
+Doing that means we don't get a Mongoose object andin return bu a plain JS Object. This brings a lot of issues like no virtual fields available and not Mongoose lifecycles etc.
+
+To give the best experience possible, we decided to rehydrate the Mongoose models, forcing us to override the Mongoose query
```

Comment:
Do we really need to explain what we did under the hood?

Reply:
Probably best to explain how it's done. Easier to understand.